### PR TITLE
Xenos now suffer when the queen dies.

### PR DESCRIPTION
**Pull request by GunHog**
# NERF ALERT!!!!

:warning: :rotating_light: :rotating_light: :rotating_light: :rotating_light: :warning: 

:cl: Gun Hog
tweak: The Alien queen, upon her death, will now severely weaken the remaining alien
forces.
/:cl:

The following changes have been requested by @WJohn:

Upon death of the queen, the remaining aliens will suffer the following effects:
- Disabled Hivemind and Evolution for **4 minutes.**
- A very brief period of slowdown.
- Stuttering.
- Confused movement, making it difficult to escape or attack.
- Jitters, as a confirmation to the crew that the aliens are at their weakest.

It comes with a new screen alert to serve as a reminder! Sprite by @WJohn!
### Intent:

The reasoning behind this pull is to drive home the importance of the queen's role and the aliens' need to defend her - especially the Praetorian. Additionally, this makes killing the queen more of the boss battle it was intended to be. Previously, another queen would simply pop up seconds after the first one died, rendering the monumental task of taking down the original queen entirely moot. 

**Defend your queen. Do not let your queen be a moron and fight alone.**

Screen alert is at the top right.

screenshot 2016-02-25 21 11 34

### CODE REVIEW

**Comment by Ultimate-Chimera** [comment]
Noooo muh xenos! ;-; :-1:

**Comment by WJohn** [comment]
:+1: 

Thank you so much for the work you put into this.

**Comment by Ultimate-Chimera** [comment]
B-but WJohn

**Comment by VincVsVincent** [comment]
This actually makes lots of sense, add it

**Comment by WJohn** [comment]
Don't worry I've got some buffs in mind for xenos.

After all, you should know by now that the development for xenos is two boo boo bandages for every nasty nerf they get!

File: code/modules/mob/living/carbon/alien/humanoid/death.dm [hunk 1]
```diff
@@ -19,3 +19,13 @@
 		mind.store_memory("Time of death: [tod]", 0)
 
 	return ..(gibbed)
+
+//When the alien queen dies, all others must pay the price for letting her die.
+/mob/living/carbon/alien/humanoid/royal/queen/death(gibbed)
+	if(stat == DEAD)
+		return
+
+	for(var/mob/living/carbon/alien/humanoid/A in living_mob_list)
```
Comment (ChangelingRain) [inline code/modules/mob/living/carbon/alien/humanoid/death.dm]: Only humanoid aliens? What about larva? Wouldn't it be possible for a larva to evolve, then be fastracked to queen via plasma transfers from other aliens to bypass this?

**Comment by Ultimate-Chimera** [comment]
Yay <3

**Comment by Ultimate-Chimera** [comment]
Make xenos evolve faster tbh @WJohn

**Comment by Core0verload** [comment]
:+1: Nice change. But you can make it display a "connection severed" message, stun, jitter and confuse everyone with a hive node, not just aliens. Then if someone used alien nodes to spy on alien chat when the queen was killed, they will be hit by the backlash too.

File: code/modules/mob/living/carbon/alien/humanoid/death.dm (hunk 1 above)
Comment (GunHog) [inline code/modules/mob/living/carbon/alien/humanoid/death.dm]: Oh, crap. I did not put it high enough. Will fix.

**Comment by GunHog** [comment]
Not a bad idea, Core. If WJ likes it, I will do it (it is his PR)!

File: code/modules/mob/living/carbon/alien/humanoid/humanoid.dm [hunk 2]
```diff
@@ -167,3 +168,23 @@ proc/alien_type_present(var/alienpath)
...
+
+	throw_alert("alien_noqueen", /obj/screen/alert/alien_vulnerable)
+	emote("roar")
+	Stun(10)
+	jitteriness += 30
+	do_jitter_animation(jitteriness)
+	confused += 30
+	stuttering += 30
+	recent_queen_death = 1
+	spawn(2400) //four minutes
```
Comment (ChangelingRain) [inline code/modules/mob/living/carbon/alien/humanoid/humanoid.dm]: Make sure the alien still exists after this long.

Comment (GunHog) [inline code/modules/mob/living/carbon/alien/humanoid/humanoid.dm]: Righto

**Comment by WJohn** [comment]
:+1: Idea, core. I think that'd be neat.
Although you'd have to make them *scream instead of *roar.

**Comment by GunHog** [comment]
Xenos do not have a sound for *scream

**Comment by WJohn** [comment]
I mean for humans, it'd need a unique exception for them. Humans don't exactly roar.

**Comment by Ultimate-Chimera** [comment]
Honestly, remove the disabled evolution because it already takes 4 minutes to evolve.

**Comment by WJohn** [comment]
It doesn't prevent larvas from growing up, it prevents drones and praetorians from evolving. That's the whole point, to prevent another queen from happening and to let the crew hunt down the remaining xenos in that time frame.

**Comment by Zilenan91** [comment]
Guess that xeno pref is staying off. Xenos were already painful as hell to play, this only makes it worse.


Also this in no way makes the Queen a boss battle because any amount of time in RnD equals laser cannons which means the queen gets 3 or 4 shot from one singular person with one singular laser cannon.

**Comment by PKPenguin321** [comment]
you cucks
"xenos are underpowered and painful to play as wah"
holy shit have you ever been a xeno

**Comment by WJohn** [comment]
You have to be awful to lose as xeno.

Like unbelievably bad to actually somehow die.

Stop being bad and you instantly win against anything and everything.

**Comment by Core0verload** [comment]
Then I think var/recent_queen_death would be better as a hive node var.

>I mean for humans, it'd need a unique exception for them.

Also exception for "Your queen has been struck down!". I don't think having a hive node makes you loyal to the hive queen.

**Comment by Zilenan91** [comment]
Oh man you have stunbaton hands. You get 4 shotted by any laser and basically one shotted by any semi-competent burn mix because you take doubled burn damage and can't put yourself out in one roll. The facts are that the crew has more bodies than you can deal with, and no matter what you do, the second they hear xenos the entire armory is passed out and your hive is instantly dead. You can do a lot of damage alone, but that tends to piss people off OOCly. All this really does is make it so you don't want to evolve to Queen or Praetorian at all because the lack of mobility does not in any way make up for what you gain in doing it.

**Comment by GunHog** [comment]
You need five shots from a laser cannon to crit a queen from full health, six if she is on weeds.

**Comment by Ultimate-Chimera** [comment]
Just make larva evolve faster or something.

**Comment by GunHog** [comment]
Also I am not sure how to go about doing the check for hive nodes without snowflaking things up.

**Comment by Zilenan91** [comment]
Making larva evolve faster would be nice. If you don't have weeds you literally have to AFK in a corner for 15 minutes to grow up one level.

**Comment by WJohn** [comment]
Faster larva evolve time would be nice, I agree. A problem with xenos is their set-up speed is really really slow because they have no weeds to stand on. Maybe we could halve the time it takes to evolve.

But zilenan, there's MANY tips available in game that tell you how to not suck as a xeno. You don't ever git hit by lasers because you're stupid if you actually put yourself in a situation where that can happen. You have stunbatton hands and leap (which is instant win) and can both abuse existing corners or *make your own battlefield thanks to drones and queens* to force melee. After that it's a matter of exposing everything to space, making liberal use of your facehuggers, and picking your targets/sabotaging the station.

**Comment by Core0verload** [comment]
>Also I am not sure how to go about doing the check for hive nodes without snowflaking things up.

```
for(/mob/living/carbon/C in living_mob_list)
    var/obj/item/organ/internal/alien/hivenode/node = getorganslot("hivenode")
    if(istype(node)) // just in case someone would ever add a diffirent node to hivenode slot
        node.recent_queen_death = 1
        C.something blah blah
```

**Comment by GunHog** [comment]
Making it REALLY hard to be lazy there, Core!

**Comment by Zilenan91** [comment]
Can't get facehuggers if it takes 5 minutes for them to grow up. And even then you're limited by the Queen who may or may not even be alive anymore due to her immobility and how many she layed in the first place because those things cost a fuckton. Also the issue with "not getting hit by lasers" doesn't exist because you can't really do anything but kill/infest as anything but a Drone, meaning you have to slowly jog towards people attempting to stun them, getting slowed by one laser shot and then killed with the rest. If you attempt to stay in your little hidey hole they're just gonna flamethrower you, or send in so many people you have no hope of winning. If you're in anywhere even remotely open or fighting more than one person you die because you can't handle that many people unless you get insane RnG. If you're alone, you can't kill alone either, because the second you start doing anything xeno-related people are going to scream over comms where you are, whereupon you get swarmed and die unless you're luckily near some vents, in which case I hope you love welded vents because all of them are now closed tighter than a gay man's anus.

Drones as a standalone are good and flexible but they can't kill people because they're so slow, do less damage, and despite all of the walls and shit they can make the walls die in 5 spear hits anyways so they basically aren't even there. You also can't use space to your advantage in any way because jetpacks exist and you literally can't deal with them since sentinels can't aim their attack, so space is offlimits, and attempting to crawl into maint rooms to grow aren't a thing either because everyone knows xenos hive in those rooms and since Queens are so immobile, if they go into a small room like electrical maint they've lost before they've even started since they'll just get swarmed out/flamethrowered, and if they make a space breach to escape maint I hope you like jetpacks because people are going to be swarming around you with them hitting you forever until you die.

Basically what I'm saying is humans have way too many options for killing you, and you're too immobile and too close ranged to ever hope to beat them. This pr takes xenos in a really shitty direction by basing them around the Queen and praetorian who die in 5 seconds the large majority of the time, rather than the supremely more effective and more fun lower castes. I've had some ideas for fixing this but I know that I'm not going to be listened to so I'm just not going to say them.

**Comment by Zilenan91** [comment]
Also even from just the concept who cares if the xenos are crippled for 4 minutes they're just gonna sit in a vent the entire time and then immediately evolve

**Comment by WJohn** [comment]
That's still a huge point for the crew. The xenos are in full retreat, you can burn away their nest with very little opposition.

**Comment by Zilenan91** [comment]
Who cares about their nest they'll just wait 5 minutes and grow it back again. The eggs are a moot point too because a drone is just gonna go off z-level and 10 minutes later they'll have more.

Actually, can I talk about this? How queens going off z-level to make their hives is more efficient than staying on the station and having conflict with people? Because it's terrible. Laying eggs is such a painfully slow process that if you're in the 10 minute gap from where you plant them and they grow up, you're down shit's creek without a paddle and will die even faster than if you actually had them because the people attacking you by this point all have biohoods which make them 100% immune to said huggers.

**Comment by AlexCrimson** [comment]
If you are playing aliens on /tg/ then you are doing it wrong. Go play on Colonial Marines instead.

Making the entire hive suffer for the shitty play of one bad queen? Terrible mechanic.

**Comment by WJohn** [comment]
I think it'd be nice if we do a bit of rebalancing to make the timers much smaller, but that's for a future project. The main point of this PR anyway is to open an attack window for the crew, because xenos can form a pretty impenetrable and slowly advancing death wall otherwise.

**Comment by Ultimate-Chimera** [comment]
Wait, we're going to see how this goes in play? <3 Tell me when! Bagil please! <3

**Comment by PKPenguin321** [comment]
Too much autism in this PR discussion to be completely honest

**Comment by Fox-McCloud** [comment]
Love it.

Can't give this enough :+1:

**Comment by Hatfish** [comment]
Since we're all complaining about xenos in this PR anyways, could someone readd shields blocking the hunter's leap?

File: code/modules/mob/living/carbon/alien/humanoid/humanoid.dm [hunk 3]
```diff
@@ -167,3 +168,23 @@ proc/alien_type_present(var/alienpath)
 			continue
 		return 1
 	return 0
+
+//When the alien queen dies, all aliens suffer a penalty as punishment for failing to protect her.
+/mob/living/carbon/alien/humanoid/proc/queen_death()
+	if(stat == DEAD)
+		return
+	src << "<span class='userdanger'>Your queen has been struck down!</span>"
```
Comment (RandomMarine) [inline code/modules/mob/living/carbon/alien/humanoid/humanoid.dm]: "Your queen has perished!" May be a better, more universal term. 'Struck down' doesn't quite fit if, for example, an accidental plasma fire killed the queen.

Comment (KazeEspada) [inline code/modules/mob/living/carbon/alien/humanoid/humanoid.dm]: I can assure you that plasma fire was no accident.

**Comment by Robustin** [comment]
If queens weren't basically a free kill I'd like it more.

The nest defenders (drones, sentinels, prae, queen) all kinda suck when the laser spam squad shows up. The insanely robust hunters meanwhile are all designed to be on the offense, and if the queen is blitzed she doesn't even have time to call for help (too busy dodging lasers).

Xenos have a real issue with this. Hunters are OP with infinite no-delay leapstun spam, everything else is target practice as long as the humans attack as a group (they can shake people up from spit or at least suppress any sentinels who move in for a tackle after a successful spit).

**Comment by WJohn** [comment]
Robustin have you played xenos in the last 6 months? Those points you raised have changed completely. You also should never have to be actively dodging lasers, you sit around corners with your tail sweep, stun hands, and neurotoxin to instawin any fight. Sentinels have been buffed such that they have more health than hunters, are at human movement speed, and are able to go invisible. Hunters also cannot pounce on crowds of people any more, they have a fairly lengthy delay between them.

**Comment by Anonmare** [comment]
There's only one buff I want for xenos

>DECREASE THE TIME IT TAKES FOR SOMEONE TO POP HOLY SHIT
At my last count, it takes someone at least 8 minutes to pop from being hugged (a little less if you kill them after the embryo takes) but the time it takes someone to burst and the slow evolution for larva without weeds is pretty ridicolous.

**Comment by WJohn** [comment]
Decreasing all these timers can be done in the future, that's not what this PR is about.

**Comment by GunHog** [comment]
WJ do not forget to leave me notes on what I should change beyond the code fixes Joan pointed out. I will get around to it when I can!

**Comment by TalkingCactus** [comment]
Slightly related: You don't have to like where it comes from but feel free to pull anything from https://github.com/TalkingCactus/Citadel-Station-13-6th-Port which includes 90% xeno changes.

**Comment by Robustin** [comment]
Just got exposed to xenos again, what a joke. 

You can add this nerf as long as my buff PR's don't get caught in a slog of idiotic comments.

>Tackle is now RNG based

Buh gawd.

**Comment by GunHog** [comment]
@Robustin I never changed the tackle ability, if you see it somewhere in my changes, point it out, as it is not intended. 

@TalkingCactus I took a look at that huge PR, and I would encourage @WJohn to have a read. All Xeno related PRs I code are on his behalf, and he has final say on what I upload (barring required code changes that do not deviate from WJ's intent).

What I find most interesting in that PR are the breathing and stomping sounds. It is something that I would love to add, but there may be concerns about stealth.

**Comment by TalkingCactus** [comment]
@GunHog I essientially switched from silent super killer death machines to loud and scary sort of robust death machines. All smaller castes gained the ability to use stealth also, which suppresses most sounds.

**Comment by GunHog** [comment]
It will be something for WJ to consider. I will probably end up coding it depending on how he feels about their balance, currently.

**Comment by TalkingCactus** [comment]
Yeah I wouldn't suggest directly porting any of it over, really. It's not optimized in the slightest and frankly I'm a noob coder.

**Comment by RemieRichards** [comment]
@TalkingCactus is the "Aliens make a noise when breathing" thing up for grabs because if done right that could be amazing.

**Comment by TalkingCactus** [comment]
@RemieRichards It's all free to use. Dig in.

File: code/modules/mob/living/carbon/alien/organs.dm [hunk 4]
```diff
@@ -118,6 +119,34 @@
...
+	owner.jitteriness += 30
+	owner.confused += 30
+	owner.stuttering += 30
+
+	recent_queen_death = 1
+	owner.throw_alert("alien_noqueen", /obj/screen/alert/alien_vulnerable)
+	spawn(2400) //four minutes
+		if(qdeleted(src) || !owner) //In case the node is deleted or removed from the host
+			return
+		recent_queen_death = 0
```
Comment (RemieRichards) [inline code/modules/mob/living/carbon/alien/organs.dm]: shouldn't this be above the check for owner?

Comment (GunHog) [inline code/modules/mob/living/carbon/alien/organs.dm]: Oh, that makes sense. I will toss that in with the (hopefully) final commit.

**Comment by RemieRichards** [comment]
been 6 days, Gun is OK for merge.
